Formula/libobjc2.rb

```diff
@@ -0,0 +1,57 @@
+class Libobjc2 < Formula
+  desc "Objective-C runtime library intended for use with Clang"
+  homepage "https://github.com/gnustep/libobjc2"
+  url "https://github.com/gnustep/libobjc2/archive/refs/tags/v2.1.tar.gz"
+  sha256 "78fc3711db14bf863040ae98f7bdca08f41623ebeaf7efaea7dd49a38b5f054c"
+  license "MIT"
+
+  depends_on "cmake" => :build
+  depends_on "pkg-config" => :test
+  # Clang explicitly forbids building Mach-O binaries of libobjc2.
+  # https://reviews.llvm.org/D46052
+  # macOS provides an equivalent Objective-C runtime.
+  depends_on :linux
+
+  uses_from_macos "llvm" => [:build, :test]
```

Comment:
Not neeeded at runtime?

Reply:
Nope - `libobcj2` does link to `libstdc++` from the host GCC for C++ interoperability but it does not use any run time libraries from LLVM like `libLLVM` etc.  Maybe I should add a comment to explain this?

Reply:
Yes, that would help. Does it work if I try to use `gcc-5` or Homebrew `gcc` in the test block as the C compiler? If not, does that mean our adding `objc[++]` to `languages` in `gcc.rb` doesn't really work on Linux?

Reply:
I'll add a comment explaining this.  I've just double checked and GCC 5, 11 and 12 do not work with the test nor can they build `gnustep-base`, so it's not an issue of GCC being too old or too new.  I think we should consider dropping `objc[++]` from GCC on all platforms as it probably does not work on macOS either.  As I mentioned above, limiting our GCC formula to just C, C++ and Fortran will make it easier to maintain on Linux, especially if we transition to always using brewed GCC as our compiler.